Adjust documentation for event system after recent API changes (#5224)
microsoft/MixedRealityToolkit-Unity: Documentation/UpdatingToGA.md

- Provides an explicit and granular control over which input events should be listened to globally and which should be focused-based.
- Allows multiple components on the same object to listen to global events independently on each other.

**How to migrate**

- If you have been calling `Register`/`Unregister` API directly before, replace these calls with calls to `RegisterHandler`/`UnregisterHandler`. Use handler interfaces you implement as generic parameters. If you implement multiple interfaces, and several of them listen to global input events, call `RegisterHandler` multiple times.
- If you have been inheriting from `InputSystemGlobalListener`, change inheritance to `InputSystemGlobalHandlerListener`. Implement `RegisterHandlers` and `UnregisterHandlers` abstract methods. In the implementation call `inputSystem.RegisterHandler` (`inputSystem.UnregisterHandler`) to register on all handler interfaces you want to listen global events for.
- If you have been inheriting from `BaseInputHandler`, implement `RegisterHandlers` and `UnregisterHandlers` abstract methods (same as for `InputSystemGlobalListener`).

**Examples of migration**

> <pre><code>

Reviewer: Consider using ```csharp instead of <pre><code>

Author: I used it to use *bold* highlight in the snippet for parts which have changes. Does it work with ``` ?

Author: Probably a bad idea though.

Since the release of RC2, there have been a number of API changes including some that may break existing projects. The following sections describe the changes that have occurred between the RC2 and GA releases.

- [Event System](#event-system)
- [Spatial Awareness](#spatial-awareness)

### Event System

**Changes**

> - The `IMixedRealityEventSystem` old API methods `Register` and `Unregister` has been marked as Obsolete. They are preserved for backward compatibility.

Reviewer 2: ```suggestion
- The `IMixedRealityEventSystem` old API methods `Register` and `Unregister` have been marked as obsolete. They are preserved for backwards compatibility.
```

- `InputSystemGlobalListener` has been marked as obsolete. Its functionality has not changed.
- `BaseInputHandler` base class has been changed from `InputSystemGlobalListener` to `InputSystemGlobalHandlerListener`. This is a breaking change for any descendants of `BaseInputHandler`.

**Motivation behind the change**

> Old event system Api `Register` and `Unregister` could potentially cause multiple issues in runtime, main being:

Reviewer 2: ```suggestion
The old event system API `Register` and `Unregister` could potentially cause multiple issues in runtime, main being:
```

> - If a component registers for global events, it would receive global input events of ALL types.

Reviewer 2: ```suggestion
- If a component registers for global events, it would receive global input events of *all* types.
```

use italic for highlighting , not uppercase

> - If one of components on an object registers for global input events, all components on this object will receive global input events of ALL types.

Reviewer 2: ```suggestion
- If one of the components on an object registers for global input events, all components on this object will receive global input events of *all* types.
```

> - If two components on the same object register to global events, and then one is disabled in runtime, second stops receiving global events.

Reviewer 2: ```suggestion
- If two components on the same object register to global events, and then one is disabled in runtime, the second one stops receiving global events.
```

> New Api `RegisterHandler` and `UnregisterHandler`:

Reviewer 2: ```suggestion
New API `RegisterHandler` and `UnregisterHandler`:
```

- Provides an explicit and granular control over which input events should be listened to globally and which should be focused-based.
- Allows multiple components on the same object to listen to global events independently on each other.

**How to migrate**

> - If you have been calling `Register`/`Unregister` API directly before, replace these calls with calls to `RegisterHandler`/`UnregisterHandler`. Use handler interfaces you implement as generic parameters. If you implement multiple interfaces, and several of them listen to global input events, call `RegisterHandler` multiple times.

Reviewer 2: can you add example code for this? will make migration easier

> - If you have been inheriting from `InputSystemGlobalListener`, change inheritance to `InputSystemGlobalHandlerListener`. Implement `RegisterHandlers` and `UnregisterHandlers` abstract methods. In implementation call `inputSystem.RegisterHandler` (`inputSystem.UnregisterHandler`) to register on all handler interfaces you want to listen global events for.

Reviewer 2: ```suggestion
- If you have been inheriting from `InputSystemGlobalListener`, change inheritance to `InputSystemGlobalHandlerListener`. Implement `RegisterHandlers` and `UnregisterHandlers` abstract methods. In the implementation call `inputSystem.RegisterHandler` (`inputSystem.UnregisterHandler`) to register on all handler interfaces you want to listen global events for.
```

can you add example code for this? will make migration easier

> - If you have been inheriting from `BaseInputHandler`, implement `RegisterHandlers` and `UnregisterHandlers` abstract methods (same as for `InputSystemGlobalListener`).

Reviewer 2: same: can you add example code for this? will make migration easier
